drivers/adc/adc_ads114s0x.c

```diff
@@ -607,6 +610,69 @@ static int ads114s0x_channel_setup(const struct device *dev,
 		return -EINVAL;
 	}
 
+	/* The ADS114 uses samples per seconds units with the lowest being 2.5SPS
+	 * and with acquisition_time only having 14b for time, this will not fit
+	 * within here for microsecond units. Use Tick units and allow the user to
+	 * specify the ODR directly.
+	 */
+	if (channel_cfg->acquisition_time != ADC_ACQ_TIME_DEFAULT &&
+	    acquisition_time_unit != ADC_ACQ_TIME_TICKS) {
+		LOG_ERR("invalid acquisition time %i", channel_cfg->acquisition_time);
+		return -EINVAL;
+	}
+
+	if (channel_cfg->acquisition_time == ADC_ACQ_TIME_DEFAULT) {
+		ADS114S0X_REGISTER_DATARATE_DR_SET(data_rate, 0b0100);
+	} else {
+		switch (acquisition_time_value) {
+		case ADS114S0X_CONFIG_DR_2_5:
+			ADS114S0X_REGISTER_DATARATE_DR_SET(data_rate, 0b0000);
```

Comment:
As you have defined the available data rates to align with the register settings, couldn't all these case entries be replaced with:
`ADS114S0X_REGISTER_DATARATE_DR_SET(data_rate, acquisition_time_value);`

Reply:
I already thought about the exact same thing, but was kind of uncertain if I should keep it in such an explicit form or not. Technically there are two possible values for 4000 SPS (0b1101 and 0b1110), for instance. But I get the point of reducing the amount of code, I will change it.

Reply:
done